furusiya/model/entities/fire.py

```diff
@@ -0,0 +1,34 @@
+import colors
+from constants import DELTA_UP, DELTA_DOWN, DELTA_LEFT, DELTA_RIGHT
+from game import Game
+from model.entities.game_object import GameObject
+from model.config import config
+
+
+class Fire(GameObject):
+    def __init__(self, x, y):
+        super().__init__(x, y, '*', 'Fire', colors.red, blocks=False)
+
+        Game.events.bind('on_entity_move', self.on_entity_move)
+        Game.events.bind('on_turn_pass', self.on_turn_passed)
+
+        self.turns_passed_alight = 0
+
+    def on_entity_move(self, entity):
+        if (entity.x, entity.y) == (self.x, self.y):
+            Game.fighter_system.get(entity).take_damage(config.data.enemies.fire.damage)
+            self.die()
+
+    def on_turn_passed(self):
+        self.turns_passed_alight += 1
+        if self.turns_passed_alight >= config.data.enemies.fire.selfExtinguishTurns:
+            self.die()
+        if config.data.enemies.fire.spreadChance >= Game.random.randint(1, 100):
+            dx, dy = Game.random.choice([DELTA_UP, DELTA_DOWN, DELTA_LEFT, DELTA_RIGHT])
+            created_fire = Fire(self.x + dx, self.y + dy)
+            Game.area_map.entities.append(created_fire)
+
+    def die(self):
+        super().die()
+        Game.events.unbind('on_entity_move', self.on_entity_move)
```

Comment:
I don't like having to manually bind/unbind events when things die. We need to discuss the design of the event bus again. I think the message bus should just delegate to `GameObject` instances which have callbacks that are called on events; the bus shouldn't store the callbacks. (Here's a good reason why.)

Reply:
If I get a chance, I will implement this in a branch off of your branch.

Reply:
Now that I think about it, your approach of an event bus keeping a list of entities might be better.

All it takes to de-register a dead entity is an all-encompassing line of `event_bus.unregister(self)`, which could be incorporated into GameObject to make this seamless.

Thoughts?